sdk/core/azure-core/src/main/java/com/azure/core/util/logging/LoggingEventBuilder.java

```diff
@@ -0,0 +1,289 @@
+     * </pre>
+     * <!-- end com.azure.core.util.logging.clientlogger.atInfo -->
+     *
+     * @param key String key.
+     * @param value String value.
+     * @return The updated {@code LoggingEventBuilder} object.
+     */
+    public LoggingEventBuilder addKeyValue(String key, String value) {
+        if (this.isEnabled) {
+            addKeyValueInternal(key, value);
+        }
+
+        return this;
+    }
+
+    /**
+     * Adds key with boolean value to the context of current log being created.
+     *
+     * @param key String key.
+     * @param value boolean value.
+     * @return The updated {@code LoggingEventBuilder} object.
+     */
+    public LoggingEventBuilder addKeyValue(String key, boolean value) {
+        if (this.isEnabled) {
+            addKeyValueInternal(key, value);
+        }
+        return this;
+    }
+
+    /**
+     * Adds key with long value to the context of current log event being created.
+     *
+     * <p><strong>Code samples</strong></p>
+     *
+     * <p>Adding an integer value to logging event context.</p>
+     *
+     * <!-- src_embed com.azure.core.util.logging.clientlogger.atverbose.addKeyValue#primitive -->
+     * <pre>
+     * logger.atVerbose&#40;&#41;
+     *     .addKeyValue&#40;&quot;key&quot;, 1L&#41;
+     *     .log&#40;&#40;&#41; -&gt; String.format&#40;&quot;Param 1: %s, Param 2: %s, Param 3: %s&quot;, &quot;param1&quot;, &quot;param2&quot;, &quot;param3&quot;&#41;&#41;;
+     * </pre>
+     * <!-- end com.azure.core.util.logging.clientlogger.atverbose.addKeyValue#primitive -->
+     *
+     * @param key String key.
+     * @param value long value.
+     * @return The updated {@code LoggingEventBuilder} object.
+     */
+    public LoggingEventBuilder addKeyValue(String key, long value) {
+        if (this.isEnabled) {
+            addKeyValueInternal(key, value);
+        }
+        return this;
+    }
+
+    /**
+     * Adds key with String value supplier to the context of current log event being created.
+     *
+     * @param key String key.
+     * @param valueSupplier String value supplier function.
+     * @return The updated {@code LoggingEventBuilder} object.
+     */
+    public LoggingEventBuilder addKeyValue(String key, Supplier<String> valueSupplier) {
+        if (this.isEnabled) {
+            if (this.context == null) {
+                this.context = new ArrayList<>();
+            }
+
+            this.context.add(new ContextKeyValuePair(key, valueSupplier));
+        }
+        return this;
+    }
+
+    /**
+     * Logs message annotated with context.
+     *
+     * @param message the message to log.
+     */
+    public void log(String message) {
+        if (this.isEnabled) {
+            logger.performLogging(level, false, getMessageWithContext(message));
+        }
+    }
+
+    /**
+     * Logs message annotated with context.
+     *
+     * @param messageSupplier string message supplier.
+     */
+    public void log(Supplier<String> messageSupplier) {
```

Comment:
ClientLogger has an additional overload for logging a message and including an exception. 
```java
log(Supplier<String> messageSupplier, Throwable throwable)
```

Reply:
yeah, I guess we can add it later? I'm onboarding amqp-core to structured logging now and my next stop is eventhubs - I'll fill the gaps as we discover them if that's ok?

Reply:
Yes, that should be fine. We can add this later, if required.